.github/workflows/codeql-analysis.yml

```diff
@@ -0,0 +1,33 @@
+name: "CodeQL"
+
+on:
+  push:
+    branches: [ main, 0.*, 1.*, next ]
+  pull_request:
+    branches: [ main ]
+  schedule:
+    - cron: '27 0 * * 4'
+
+jobs:
+  analyze:
+    name: Analyze
+    runs-on: ubuntu-latest
+    permissions:
+      actions: read
+      contents: read
+      security-events: write
+    strategy:
+      fail-fast: false
+      matrix:
+        language: [ 'go' ]
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v2
```

Comment:
Shouldnt this intend to scan PR that is being merged to avoid overhead ?

Reply:
I'm not sure what you mean? note: this file is the default they provide.

Reply:
What I mean this policy should be applied to PR merge only, part of CI configuration.

We are checking in default, how about just use default  directly to avoid unnessary maintenance.

Reply:
> What I mean this policy should be applied to PR merge only, part of CI configuration.

Why should this be applied only to PR merges, and not to all refs?

Reply:
Is this about having it on both PRs and on cron? That seems reasonable, since it can catch existing issues already in the codebase and ones that would be introduced by new PRs. It looks like CodeQL's team will continuously add new patterns to their database, so it makes sense to re-run this periodically on existing code to catch previously-unknown vulnerabilities.

Using it on both push and PR events may be superfluous, as we PR for any branch that we care about (main, next, and major+minor version branches) rather than pushing directly.